### PR TITLE
More revisions and additions to the documentation

### PR DESCRIPTION
**Pull request by chudur-budur**
More revisions and additions to the documentation.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: docs/source/user_guide/config.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.. include:: ./../ext_links.txt
+
+Configuration Options for ``numba-dpex``
+========================================
+
+``numba-dpex`` provides a set of environment variables and flags for configuring different aspects of the compilation, debugging and execution of programs. The configuration flags of ``numba-dpex`` are mostly inherited from those of Numba*. They are defined in :file:`numba_dpex/core/config.py` and are listed as follows:
+
+``SAVE_IR_FILES``:
```
Comment (diptorupd) [inline docs/source/user_guide/config.rst:8]: It should be mentioned that when setting these configuration variables using environment variables, the name of the environment variable is `NUMBA_DPEX_<name of config>`

Comment (chudur-budur) [reply docs/source/user_guide/config.rst:8]: I explained it on line 41.

Comment (diptorupd) [reply docs/source/user_guide/config.rst:8]: Can we move the note on top of the list? That way no one should miss it.

Comment (chudur-budur) [reply docs/source/user_guide/config.rst:22]: > Can we move the note on top of the list? That way no one should miss it.

Fixed.

**Pull request merged**